OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs

```diff
@@ -479,7 +493,36 @@ protected override void Dispose(bool disposing)
 				Game.BeforeGameStart -= RemoveShellmapUI;
 			}
 
+			Game.OnShellLoaded -= OpenMenuBasedOnLastGame;
 			base.Dispose(disposing);
 		}
+
+		public void OpenMenuBasedOnLastGame()
```

Comment:
Style nit: use the lowest level of visibility required to get the job done - remove the `public` here.